canonical/assets.ubuntu.com
# feat: Implement UI for create and update

## Done

- Creates a new template that will be used for both 'create' and 'update', create-update.html`. The fields are disabled and enabled based on whether they are editable in each state, 'create' and 'update'.
- When the form is rendered for 'update', the data for the relevant asset is passed to the template and the form is pre-filled with existing asset information. This is accomplished partly using Jinja and partly with JS

## QA

- Check out this feature branch and run the project
- Navigate to the 'Add asset' tab (http://0.0.0.0:8017/manager/create) and attempt to upload an asset, making sure to test each field. It would be a good idea to test with different combinations of fields and, when relevant, test with different amounts of inputs ie. on tags.
- Navigate to the asset update page, (http://0.0.0.0:8017/manager/update?file-path=FILE_PATH), and see that the fields you previously selected/input are already filled in. Update some fields and click update, see you are redirected to the same page but with the relevant fields updated.
- Make sure it matches the design/fields/descriptions ect in the [figma](https://www.figma.com/design/dOz7GIHMmofSdje0l0pVFs/Asset-manager?node-id=12-4914&node-type=frame&t=rS6m8vOfxnAr0WK0-0) and [spec](https://docs.google.com/document/d/1QzAjI0OfjagcuGtfZpKv2h31tyAhM4-cjL0xa-2dXW8/edit)

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-13751
